### ISSUE TITLE
Document `jax.lax.padtype_to_pads`

### ISSUE THREAD

**Issue by patrick-kidger** [google/jax#19874]
JAX has the `jax.lax.padtype_to_pads` function for converting string-specified padding e.g. `padding='same'` into the before-and-after numeric padding.

This [seems to be useful](https://github.com/patrick-kidger/equinox/pull/658) for downstream libraries looking to implement their own convolutional layers. In particular this comes up when wanting to (a) duplicate the string-based padding behaviour of `lax.conv_general_dilated`, but (b) use nonzero padding (e.g. circular padding instead).

Right now it's not documented. Would this be something we can add to the public API?

**Comment by Aniketsy** [comment]
Hi @patrick-kidger , I would like to work on this issue, if I have your permission. I’d also appreciate any guidance on where to begin (e.g., which files or docs should be updated). Thanks!
